smooth-app-bar-layout/src/main/java/me/henrytao/smoothappbarlayout/SmoothAppBarLayout.java

```diff
@@ -151,6 +151,7 @@ public int getCurrentOffset() {
 
   public void setScrollTargetCallback(ScrollTargetCallback scrollTargetCallback) {
     mScrollTargetCallback = scrollTargetCallback;
+    setBehaviorScrollTargetCallback();
```

Comment:
I think keeping this method is not a right design because `setScrollTargetCallback` is preferred to call in onCreate and should not be change in other states of Activity / Fragment lifecycle which mean calling this method is redundant. With the update yesterday, whenever `mScrollTargetCallback` is set, it will prioritize the result from this method first. So that, it's more flexible now. 

```
return mScrollTargetCallback != null ? mScrollTargetCallback.callback(target) : getSupportedScrollTarget(target);
```
I prefer not to keep `setBehaviorScrollTargetCallback` method and rollback some changes in `Behavior` class. The rest looks good to merge.